integration-tests/smoke/vrfv2plus_test.go

```diff
@@ -155,4 +134,144 @@ func TestVRFv2PlusBilling(t *testing.T) {
+		require.NoError(t, err, "error registering migratable coordinator")
+
+		err = env.EVMClient.WaitForEvents()
+		require.NoError(t, err, vrfv2plus.ErrWaitTXsComplete)
+
+		err = vrfv2PlusContracts.Coordinator.Migrate(subID, newCoordinator.Address())
+		require.NoError(t, err, "error migrating sub id ", subID.String(), " from ", vrfv2PlusContracts.Coordinator.Address(), " to new Coordinator address ", newCoordinator.Address())
+		migrationCompletedEvent, err := vrfv2PlusContracts.Coordinator.WaitForMigrationCompletedEvent(time.Minute * 1)
+		require.NoError(t, err, "error waiting for MigrationCompleted event")
+		err = env.EVMClient.WaitForEvents()
+		require.NoError(t, err, vrfv2plus.ErrWaitTXsComplete)
+
+		l.Debug().
+			Str("Subscription ID", migrationCompletedEvent.SubId.String()).
+			Str("Migrated From Coordinator", vrfv2PlusContracts.Coordinator.Address()).
+			Str("Migrated To Coordinator", migrationCompletedEvent.NewCoordinator.String()).
+			Msg("MigrationCompleted Event")
+
+		migratedSubscription, err := newCoordinator.GetSubscription(context.Background(), subID)
+		require.NoError(t, err, "error getting subscription information")
+
+		l.Debug().
+			Interface("New Coordinator", newCoordinator.Address()).
+			Interface("Subscription ID", subID).
+			Interface("Juels Balance", migratedSubscription.Balance).
+			Interface("Native Token Balance", migratedSubscription.EthBalance).
+			Interface("Subscription Owner", migratedSubscription.Owner.String()).
+			Interface("Subscription Consumers", migratedSubscription.Consumers).
+			Msg("Subscription Data After Migration to New Coordinator")
+
+		//Verify that Coordinators were updated in Consumers
+		for _, consumer := range consumersForMigration {
+			coordinatorAddressInConsumerAfterMigration, err := consumer.GetCoordinator(context.Background())
+			require.NoError(t, err, "error getting Coordinator from Consumer contract")
+			require.Equal(t, newCoordinator.Address(), coordinatorAddressInConsumerAfterMigration.String())
+			l.Debug().
+				Interface("Consumer", consumer.Address()).
+				Interface("Coordinator", coordinatorAddressInConsumerAfterMigration).
+				Msg("Coordinator Address in Consumer After Migration")
+		}
+
+		//Verify old and migrated subs
+		require.Equal(t, oldSubscription.EthBalance, migratedSubscription.EthBalance)
+		require.Equal(t, oldSubscription.Balance, migratedSubscription.Balance)
+		require.Equal(t, oldSubscription.Owner, migratedSubscription.Owner)
+		require.Equal(t, oldSubscription.Consumers, migratedSubscription.Consumers)
+
+		//Verify that old sub was deleted from old Coordinator
+		oldSubscription, err = vrfv2PlusContracts.Coordinator.GetSubscription(context.Background(), subID)
+		require.Error(t, err, "error not occurred when trying to get deleted subscription from old Coordinator after sub migration")
+
+		//Verify rand requests fulfills with Link Token billing
```

Comment:
would be good to validate that `s_totalBalance` and `s_totalEThBalance` was incremented as expected